### PR TITLE
[Actions] PHP 8.2 Tests & Checks

### PR DESCRIPTION
**Pull request by brusch**

### CODE REVIEW

**Comment by jdreesen** [comment]
Will Pimcore 10 also be compatible to PHP 8.2?

**Comment by brusch** [comment]
@jdreesen currently we don't have this on our list, no 😬

**Comment by jdreesen** [comment]
I'm just asking because [the `composer.json` suggests it is compatible](https://github.com/pimcore/pimcore/blob/10.5/composer.json#L31) (I haven't tried it, though).

**Comment by brusch** [comment]
@jdreesen yep, noticed that as well 👍😉  We need to check and see, if it's easily possible I'd say it would be ok for 10.6.

File: models/DataObject/ClassDefinition/Data/Extension/ColumnType.php [hunk 1]
```diff
@@ -18,16 +18,14 @@
 
 trait ColumnType
 {
+    public $columnType = null;
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/Extension/ColumnType.php]:
```suggestion
    public array|string|null $columnType = null;
```

File: models/DataObject/ClassDefinition/Data/Extension/ColumnType.php [hunk 2]
```diff
@@ -37,9 +35,7 @@ public function getColumnType(): array|string|null
      */
     public function setColumnType($columnType): static
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/Extension/ColumnType.php]:
```suggestion
    public function setColumnType(array|string $columnType): static
```

File: models/DataObject/ClassDefinition/Data/Extension/QueryColumnType.php [hunk 3]
```diff
@@ -18,16 +18,14 @@
 
 trait QueryColumnType
 {
+    public $queryColumnType = null;
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/Extension/QueryColumnType.php]:
```suggestion
    public array|string|null $queryColumnType = null;
```

File: models/DataObject/ClassDefinition/Data/Extension/QueryColumnType.php [hunk 4]
```diff
@@ -37,9 +35,7 @@ public function getQueryColumnType(): array|string|null
      */
     public function setQueryColumnType($queryColumnType): static
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/Extension/QueryColumnType.php]:
```suggestion
    public function setQueryColumnType(array|string $queryColumnType): static
```

**Review by blankse — CHANGES_REQUESTED** [review]
Add type hints

**Comment by brusch** [comment]
Depends on #13888

**Comment by kingjia90** [comment]
Closing this PR in favor of https://github.com/pimcore/pimcore/pull/14080
As requested here https://github.com/pimcore/pimcore/pull/14080#issuecomment-1398099768

**Pull request closed without merging**